### PR TITLE
Manually handle HTTP 103 through WebView

### PR DESCRIPTION
**Pull request by AntsyLich**
Related to #6623

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    internal class JsInterface(private val latch: CountDownLatch, var payload: String? = null) {
+        @JavascriptInterface
+        fun passPayload(passedPayload: String) {
+            payload = passedPayload
+            latch.countDown()
+        }
+    }
+
+    private fun http103Exception(msg: String) = Exception("Http103Interceptor - $msg")
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:82]: Should be an `IOException` considering it's in an interceptor.

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            return@lazy
+        }
+
+        WebSettings.getDefaultUserAgent(context)
+    }
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+        if (response.code != 103) return response
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:56]: I feel like we might want to just handle all 1xx responses in this interceptor. Thoughts?

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    private val executor = ContextCompat.getMainExecutor(context)
+
+    private val networkHelper: NetworkHelper by injectLazy()
+
+    /**
+     * When this is called, it initializes the WebView if it wasn't already. We use this to avoid
+     * blocking the main thread too much. If used too often we could consider moving it to the
+     * Application class.
+     */
+    private val initWebView by lazy {
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:42]: We can definitely dedupe some of this stuff, but I'll refactor that afterwards.

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        exception?.let { throw it }
+
+        val payload = jsInterface.payload ?: throw http103Exception("Couldn't fetch site through webview")
+
+        return ogResponse.newBuilder()
+            .code(200)
+            .protocol(Protocol.HTTP_1_1)
+            .message("SUCCESS")
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:154]: Is a message even required?

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt (hunk 2 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:56]: imo others doesn't impact tachi that much. 100 is handled by okhttp and 102 is mainly used by WebDAV, and 101 indicates switch protocol which should just be handled by the ext itself

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt (hunk 4 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:154]: not sure but no harm in keeping it

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt (hunk 1 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:82]: All exception are wrapped in `IOException` in intercept function

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt (hunk 2 above)
Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:56]: 1xx responses should've had no effect at all but here we are. I guess we can address it again if they decide to add more codes.

File: app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt (hunk 4 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/tachiyomi/network/interceptor/Http103Interceptor.kt:154]: > But these messages are something called HTTP status messages. In fact, the server always returns a message for every request. 

Guess that's that

**Pull request merged**